### DISCUSSION TITLE
How do I change generated local address url in port forwarding?

### DISCUSSION THREAD

**Discussion by Wireless4024** [coder/code-server#5976] [Q&A] [answered]
I'm using code-server in coder, 
by default auto forward will generate local address such as `http://localhost:{PORT}`, coder have config to use wildcard subdomain but I don't want to use it.
so I want to change it to be something like `{CURRENT_URL}/proxy/{PORT}` instead, how do I achieve this?  (url that I want is accessible if I type it manually)

if this is coder's problem or have soluion to do please let me know.

**Comment by jsjoeio** [comment, marked as answer]
Great question! Apologies for not having this in the docs or easier to find. This is where it's documented: https://github.com/coder/code-server/releases/tag/v4.8.0

So you'll want to set `VSCODE_PROXY_URI=https://your-url/proxy/{{port}}` as an environment variable. Let me know if you have any issues!

**Reply by michaelkoelle** [reply]
Unfortunately, this does not work for me. I am running coder via the kubernetes helm chart. When i set VSCODE_PROXY_URI=https://your-url/proxy/{{port}}. The local address is still https://your-url/. I double checked that the env var is set in the coder container.

**Reply by code-asher** [reply]
Hmm it seems to work as expected for me.  Is the environment variable for sure set in code-server's environment?

Setup:

```
$ /tmp/code-server/bin/code-server --version 
4.20.0 d0215eca8804d0be5d3551c39404e336032c3141 with Code 1.85.1
$ VSCODE_PROXY_URI=https://your-url/proxy/{{port}} /tmp/code-server/bin/code-server
[2024-01-05T20:05:22.993Z] info  code-server 4.20.0 d0215eca8804d0be5d3551c39404e336032c3141
[2024-01-05T20:05:22.994Z] info  Using user-data-dir /home/coder/.local/share/code-server
[2024-01-05T20:05:23.006Z] info  Using config file /home/coder/.config/code-server/config.yaml
[2024-01-05T20:05:23.007Z] info  HTTP server listening on http://127.0.0.1:8080/
[2024-01-05T20:05:23.007Z] info    - Authentication is enabled
[2024-01-05T20:05:23.007Z] info      - Using password from /home/coder/.config/code-server/config.yaml
[2024-01-05T20:05:23.007Z] info    - Not serving HTTPS
[2024-01-05T20:05:23.007Z] info  Using proxy URI in PORTS tab: https://your-url/proxy/{{port}}
[2024-01-05T20:05:23.007Z] info  Session server listening on /home/coder/.local/share/code-server/code-server-ipc.sock
```

In code-server's terminal:

```
$ printenv VSCODE_PROXY_URI
https://your-url/proxy/{{port}}
$ python3 -m http.server 8000
Serving HTTP on 0.0.0.0 port 8000 (http://0.0.0.0:8000/) ...
```

And the result in the ports panel:

![ports](https://github.com/coder/code-server/assets/45609798/9c0ab474-95d3-45da-9fd4-3b1ef46d862b)

**Reply by michaelkoelle** [reply]
Thank you, i was setting the environment variable in my coder helm values. Instead i should have set it in the template within coder and now it works. I was trying to get port-forwarding without a wildcard cert to work, but currently it can only be done via subdomains. From the doc:

> To enable port forwarding via the dashboard, Coder must be configured with a [wildcard access URL](https://coder.com/docs/v2/latest/admin/configure#wildcard-access-url).

**Reply by br-barrowkwan** [reply]
@michaelkoelle 
I am having the same issue.  do you mind to share what you have to add in the template? thanks

**Reply by michaelkoelle** [reply]
> @michaelkoelle I am having the same issue. do you mind to share what you have to add in the template? thanks

I edited the default kubernetes terraform template and added the env variable like this: 

```
env {
    name  = "VSCODE_PROXY_URI"
    value = "https://your-url/proxy/{{port}}/"
}
```

Code-server now has the correct link under the forwarded ports section. Unfortunately, coder does not know about this route and I could not access the container via the link.

**Reply by patbec** [reply]
Hi,
I think the value in `VSCODE_PROXY_URI` is missing a trailing `/` at the end.

It should look like this:  
`https://your-url/proxy/{{port}}/`

When I start the python server with `python3 -m http.server 8000` on a folder and open a file with the url, i am redirected to:
`https://your-url/proxy/8000/helloworld.html`.

Without the `/` at the end, the port is not recognized as a uri path segment and `https://your-url/proxy/hello.html` is called, with an _invalid port_ message from code-server.

I don't know if that has any side effects.

**Reply by michaelkoelle** [reply]
ty, i updated the code above
